### PR TITLE
Documentation cleanup/fixes

### PR DESCRIPTION
**Pull request by localden**

### CODE REVIEW

File: msal-dotnet-articles/acquiring-tokens/overview.md [hunk 1]
```diff
@@ -14,65 +14,54 @@ ms.reviewer: localden
...
-- On .NET framework desktop client applications , it's also possible (but not recommended) to [get a token with a username and password](./desktop-mobile/username-password-authentication.md) (U/P).
-  > Note that you should not use username/password in confidential client applications.
-- Finally, for applications running on devices which don't have a Web browser, it's possible to acquire a token through the [device code flow](./desktop-mobile/device-code-flow.md) mechanism, which provides the user with a URL and a code. The user goes to a web browser on another device, enters the code and signs-in, which has Microsoft Entra ID to get them a token back on the browser-less device.
+- Will often [acquire token interactively](./desktop-mobile/acquiring-tokens-interactively.md), having the user sign-in.
+- It's also possible for a desktop application running on a Windows machine joined to a domain or to Microsoft Entra ID to [use Integrated Windows Authentication (IWA/Kerberos)](./desktop-mobile/integrated-windows-authentication.md) to acquire a token silently.
```
Comment (pmaytak) [inline msal-dotnet-articles/acquiring-tokens/overview.md:35]: Mention that IWA is not recommended.

File: msal-dotnet-articles/acquiring-tokens/overview.md [hunk 2]
```diff
@@ -14,65 +14,54 @@ ms.reviewer: localden
...
+- For .NET Framework desktop applications, in limited scenarios it's possible to [get a token with a username and password](./desktop-mobile/username-password-authentication.md). Due to security considerations, this approach is not recommended.
+- Applications running on devices which don't have a web browser, a token can be acquired with the help of the [device code flow](./desktop-mobile/device-code-flow.md), which provides the application user with a URL and a code. The user will subsequently go to a web browser on another device, enter the code, and sign in. The authenticating device will then poll Microsoft Entra ID services until it receives confirmation of a successful sign in and an access token.
```
Comment (pmaytak) [inline msal-dotnet-articles/acquiring-tokens/overview.md:37]:
```suggestion
- In applications running on devices which don't have a web browser, a token can be acquired with the help of the [device code flow](./desktop-mobile/device-code-flow.md), which provides the application user with a URL and a code. The user will subsequently go to a web browser on another device, enter the code, and sign in. The authenticating device will then poll Microsoft Entra ID services until it receives confirmation of a successful sign in and an access token.
```

File: msal-dotnet-articles/acquiring-tokens/overview.md [hunk 3]
```diff
@@ -14,65 +14,54 @@ ms.reviewer: localden
...
+| Operating system      | Platform        | App type | [Interactive](./desktop-mobile/acquiring-tokens-interactively.md) | [IWA](./desktop-mobile/integrated-windows-authentication.md) | [ROPC](./desktop-mobile/username-password-authentication.md) | [Device Code](./desktop-mobile/device-code-flow.md) |
+|:----------------------|:----------------|:---------|:------------------------------------------------------------------|:-------------------------------------------------------------|:-------------------------------------------------------------|:----------------------------------------------------|
+| Windows desktop	      | .NET            | Desktop (WPF, Windows Forms, Console) | ✅ | ✅ | ✅ | ✅ |
+| Windows 10	          | UWP	            | Store app	                            | ✅	| ✅	| ❌	| ❌ |
+| Android	              | Xamarin Android	| Mobile	                              | ✅	| ❌	| ✅	| ❌ |
+| iOS                   | Xamarin iOS	    | Mobile	                              | ✅	| ❌	| ❌	| ✅ |
```
Comment (pmaytak) [inline msal-dotnet-articles/acquiring-tokens/overview.md:46]: Device code not available on Android? ROPC - not on iOS? Is that right?

File: msal-dotnet-articles/acquiring-tokens/overview.md [hunk 4]
```diff
@@ -14,65 +14,54 @@ ms.reviewer: localden
...
-Windows | <img alt=".NET Logo" src="/azure/active-directory/develop/media/index/logo_net.svg" width="40"/> <br/> .NET Framework | ![Web API icon](../media/web-api-icon.png) | Y | Y |
-Windows, MacOS, Linux | <img alt=".NET Core Logo" src="/azure/active-directory/develop/media/index/logo_netcore.svg" width="40" /><br/> ASP.NET Core | ![Web API icon](../media/web-api-icon.png) | Y | Y |
-Windows | <img alt=".NET Logo" src="/azure/active-directory/develop/media/index/logo_net.svg" width="40"/> <br/> .NET Framework | ![Daemon icon](../media/daemon-icon.png) <br/> (windows service) | Y | |
-Windows, MacOS, Linux | <img alt=".NET Core Logo" src="/azure/active-directory/develop/media/index/logo_netcore.svg" width="40" /> <br/> .NET Core | ![Daemon icon](../media/daemon-icon.png)| Y | |
+| Operating system      | Platform       | App type | [Client Credential](./web-apps-apis/client-credential-flows.md) | [On behalf of](./web-apps-apis/on-behalf-of-flow.md) | [Auth Code](./web-apps-apis/authorization-codes.md)
```
Comment (pmaytak) [inline msal-dotnet-articles/acquiring-tokens/overview.md:57]:
```suggestion
| Operating system      | Platform       | App type | [Client Credential](./web-apps-apis/client-credential-flows.md) | [On-Behalf-Of](./web-apps-apis/on-behalf-of-flow.md) | [Auth Code](./web-apps-apis/authorization-codes.md)
```

File: msal-dotnet-articles/acquiring-tokens/overview.md [hunk 5]
```diff
@@ -14,65 +14,54 @@ ms.reviewer: localden
...
+- For web APIs calling an API on behalf of a user, developers can use [On Behalf Of flow](./web-apps-apis/on-behalf-of-flow.md). The application itself will use client credentials to acquire a token based on a user assertion (e.g., [SAML](/entra/identity-platform/single-sign-on-saml-protocol) or a [JWT](/entra/identity-platform/security-tokens#json-web-tokens-and-claims)). This flow can be used for applications which need to access resources of a particular user in service-to-service calls.
+- **For web apps**, token acquisition is done with the help of an [by authorization code](./web-apps-apis/authorization-codes.md) after signing the user in through the authorization request URL. This is typically the mechanism used by an application which lets the user sign-in using [OpenID Connect](https://openid.net/developers/how-connect-works/), but then wants to access web APIs on behalf of this particular user.
```
Comment (pmaytak) [inline msal-dotnet-articles/acquiring-tokens/overview.md:53]:
```suggestion
- **For web apps**, token acquisition is done using an [authorization code](./web-apps-apis/authorization-codes.md) after signing the user in through the authorization request URL. This is typically the mechanism used by an application which lets the user sign-in using [OpenID Connect](https://openid.net/developers/how-connect-works/) and then accesses web APIs on behalf of this particular user.
```

File: msal-dotnet-articles/acquiring-tokens/overview.md [hunk 6]
```diff
@@ -14,65 +14,54 @@ ms.reviewer: localden
...
-Windows, MacOS, Linux | <img alt=".NET Core Logo" src="/azure/active-directory/develop/media/index/logo_netcore.svg" width="40" /> <br/> .NET Core | ![Daemon icon](../media/daemon-icon.png)| Y | |
+| Operating system      | Platform       | App type | [Client Credential](./web-apps-apis/client-credential-flows.md) | [On behalf of](./web-apps-apis/on-behalf-of-flow.md) | [Auth Code](./web-apps-apis/authorization-codes.md)
+|:----------------------|:---------------|:---------|:----------------------------------------------------------------|:-----------------------------------------------------|:---------------------------------------------------
+| Windows               | .NET Framework | Web app                  | ✅ | ❌ | ✅ |
+| Windows, macOS, Linux | ASP.NET Core   | Web app                  | ✅ | ❌ | ✅ |
+| Windows               | .NET Framework | Web API                  | ✅ | ✅ | ❌ |
+| Windows, macOS, Linux | ASP.NET Core   | Web API                  | ✅ | ✅ | ❌ |
+| Windows               | .NET Framework | Daemon (windows service) | ✅ | ❌ | ❌ |
```
Comment (pmaytak) [inline msal-dotnet-articles/acquiring-tokens/overview.md:63]:
```suggestion
| Windows               | .NET Framework | Daemon (Windows service) | ✅ | ❌ | ❌ |
```

**Review by pmaytak — APPROVED** [review]

File: msal-dotnet-articles/acquiring-tokens/overview.md [hunk 7]
```diff
@@ -14,65 +14,55 @@ ms.reviewer: localden
...
+| Operating system      | Platform        | App type | [Interactive](./desktop-mobile/acquiring-tokens-interactively.md) | [IWA](./desktop-mobile/integrated-windows-authentication.md) | [ROPC](./desktop-mobile/username-password-authentication.md) | [Device Code](./desktop-mobile/device-code-flow.md) |
+|:----------------------|:----------------|:---------|:------------------------------------------------------------------|:-------------------------------------------------------------|:-------------------------------------------------------------|:----------------------------------------------------|
+| Windows (desktop)     | .NET            | Desktop (WPF, Windows Forms, Console) | ✅ | ✅ | ✅ | ✅ |
+| Windows 10 and newer  | UWP             | Store app                             | ✅ | ✅ | ❌ | ❌ |
+| Android               | Xamarin Android | Mobile                                | ✅ | ❌ | ✅ | ❌ |
+| iOS                   | Xamarin iOS     | Mobile                                | ✅ | ❌ | ❌ | ✅ |
```
Comment (pmaytak) [inline msal-dotnet-articles/acquiring-tokens/overview.md:47]: This matches the previous table, but wouldn't Device Code be available on all; and ROPC either available or not on both, Android and iOS? @bgavrilMS

File: msal-dotnet-articles/acquiring-tokens/overview.md [hunk 8]
```diff
@@ -14,65 +14,55 @@ ms.reviewer: localden
...
-Windows, MacOS, Linux | <img alt=".NET Core Logo" src="/azure/active-directory/develop/media/index/logo_netcore.svg" width="40" /><br/> ASP.NET Core | ![Web API icon](../media/web-api-icon.png) | Y | Y |
-Windows | <img alt=".NET Logo" src="/azure/active-directory/develop/media/index/logo_net.svg" width="40"/> <br/> .NET Framework | ![Daemon icon](../media/daemon-icon.png) <br/> (windows service) | Y | |
-Windows, MacOS, Linux | <img alt=".NET Core Logo" src="/azure/active-directory/develop/media/index/logo_netcore.svg" width="40" /> <br/> .NET Core | ![Daemon icon](../media/daemon-icon.png)| Y | |
+| Operating system      | Platform       | App type | [Client Credential](./web-apps-apis/client-credential-flows.md) | [On-Behalf-Of](./web-apps-apis/on-behalf-of-flow.md) | [Auth Code](./web-apps-apis/authorization-codes.md)
+|:----------------------|:---------------|:---------|:----------------------------------------------------------------|:-----------------------------------------------------|:---------------------------------------------------
+| Windows               | .NET Framework | Web app                  | ✅ | ❌ | ✅ |
```
Comment (pmaytak) [inline msal-dotnet-articles/acquiring-tokens/overview.md:60]: These icons are nice. Very readable.

File: msal-dotnet-articles/acquiring-tokens/web-apps-apis/client-credential-flows.md [hunk 9]
```diff
@@ -1,130 +1,132 @@
...
-                   .ExecuteAsync();
+// When making the request, specify the tenant-based authority
+var authResult = await app.AcquireTokenForClient(scopes: new [] {  "some_app_id_uri/.default"})        // Uses the token cache automatically, which is optimized for multi-tenant access
+        .WithAuthority(AzureCloudInstance.AzurePublic, "{tenantID}")  // Do not use "common" or "organizations"!
+        .ExecuteAsync();
 ```
 
-**Important: do not use `common` or `organizations` authority for client credential flows.**
+>[!IMPORTANT]
+>Do not use `common` or `organizations` authority for client credential flows.
```
Comment (pmaytak) [inline msal-dotnet-articles/acquiring-tokens/web-apps-apis/client-credential-flows.md:45]:
```suggestion
>Do not use `common` or `organizations` authority for client credential flows. Specify the tenant ID in the authority.
```

File: msal-dotnet-articles/how-to/token-cache-serialization.md [hunk 10]
```diff
@@ -16,11 +16,19 @@ ms.custom: devx-track-csharp, aaddev, has-adal-ref, devx-track-dotnet
...
+# Token cache serialization
 
 After Microsoft Authentication Library (MSAL) [acquires a token](/azure/active-directory/develop/msal-acquire-cache-tokens), it caches that token. Public client applications (desktop and mobile apps) should try to get a token from the cache before acquiring a token by another method. Acquisition methods on confidential client applications manage the cache themselves. This article discusses default and custom serialization of the token cache in MSAL.NET.
 
-## Quick summary
+## Token cache types
+
+MSAL.NET operates with two types of token caches - **user** and **application**.
+
+The **application token cache** which holds access tokens for this application. It's maintained and updated silently when calling [AcquireTokenForClient](xref:Microsoft.Identity.Client.ConfidentialClientApplication.AcquireTokenForClient(System.Collections.Generic.IEnumerable{System.String})).
```
Comment (pmaytak) [inline msal-dotnet-articles/how-to/token-cache-serialization.md:27]:
```suggestion
The **application token cache** holds access tokens for this application. It's maintained and updated silently when calling [AcquireTokenForClient](xref:Microsoft.Identity.Client.ConfidentialClientApplication.AcquireTokenForClient(System.Collections.Generic.IEnumerable{System.String})).
```

File: msal-dotnet-articles/how-to/token-cache-serialization.md [hunk 11]
```diff
@@ -16,11 +16,19 @@ ms.custom: devx-track-csharp, aaddev, has-adal-ref, devx-track-dotnet
 #Customer intent: As an application developer, I want to learn about token cache serialization so I can have fine-grained control of the proxy.
 ---
 
-# Token cache serialization in MSAL.NET
+# Token cache serialization
 
 After Microsoft Authentication Library (MSAL) [acquires a token](/azure/active-directory/develop/msal-acquire-cache-tokens), it caches that token. Public client applications (desktop and mobile apps) should try to get a token from the cache before acquiring a token by another method. Acquisition methods on confidential client applications manage the cache themselves. This article discusses default and custom serialization of the token cache in MSAL.NET.
 
-## Quick summary
+## Token cache types
```
Comment (pmaytak) [inline msal-dotnet-articles/how-to/token-cache-serialization.md:23]: I would put this section at the bottom of the page. The token cache types are ideally abstracted away from the developers when either the static cache or Identity Web caches are used, and only matters when custom caching is used, which is not recommended anyway.

**Review by pmaytak — APPROVED** [review]
Minor suggestions and an outstanding question about that table.

File: msal-dotnet-articles/acquiring-tokens/web-apps-apis/client-credential-flows.md (hunk 9 above)
Comment (trwalke) [reply msal-dotnet-articles/acquiring-tokens/web-apps-apis/client-credential-flows.md:45]: "authorities"?

File: msal-dotnet-articles/acquiring-tokens/web-apps-apis/client-credential-flows.md [hunk 12]
```diff
@@ -1,130 +1,132 @@
...
 
-Check our [sample](https://github.com/Azure-Samples/active-directory-dotnet-v1-to-v2/blob/b48c10180665260a1aec78a9acf7d1b1ff97e5ba/ConfidentialClientTokenCache/Program.cs) to see how token cache serialization works.
+### Service is running out of memory
 
-## High Availability
+See [Using MSAL.NET for client credential flow in multi-tenant services](../../advanced/client-credential-multi-tenant.md) for an in-depth overview of the multi-tenant architecture with MSAL.NET.
 
-**Problem:**
-My service is running out of memory.
+Make sure to provision enough RAM on the machines running your service or use a distributed cache. A single token is a a few kilobytes (KB) in size, and one token is stored for each tenant with which the application interacts.
```
Comment (trwalke) [inline msal-dotnet-articles/acquiring-tokens/web-apps-apis/client-credential-flows.md:66]:
```suggestion
Make sure to provision enough RAM on the machines running your service or use a distributed cache. A single token is a few kilobytes (KB) in size, and one token is stored for each tenant with which the application interacts.
```

**Review by trwalke — APPROVED** [review]

**Review by trwalke — APPROVED** [review]

File: msal-dotnet-articles/how-to/token-cache-serialization.md [hunk 13]
```diff
@@ -16,11 +16,19 @@ ms.custom: devx-track-csharp, aaddev, has-adal-ref, devx-track-dotnet
...
+
+MSAL.NET operates with two types of token caches - **user** and **application**.
+
+The **application token cache** holds access tokens for this application. It's maintained and updated silently when calling [AcquireTokenForClient](xref:Microsoft.Identity.Client.ConfidentialClientApplication.AcquireTokenForClient(System.Collections.Generic.IEnumerable{System.String})).
```
Comment (neha-bhargava) [inline msal-dotnet-articles/how-to/token-cache-serialization.md:27]:
```suggestion
The **application token cache** holds access tokens and refresh tokens for an application. It's maintained and updated silently when calling [AcquireTokenForClient](xref:Microsoft.Identity.Client.ConfidentialClientApplication.AcquireTokenForClient(System.Collections.Generic.IEnumerable{System.String})).
```

**Review by neha-bhargava — APPROVED** [review]

**Pull request merged**